packages/compass-e2e-tests/tests/assistant.test.ts

```diff
@@ -0,0 +1,581 @@
+      .getAttribute('aria-checked')) === 'true';
+
+  if (currentValue !== newValue) {
+    await browser.clickParent(
+      Selectors.SettingsInputElement('enableGenAIFeatures')
+    );
+    await browser.clickVisible(Selectors.SaveSettingsButton);
+  }
+
+  const closeButton = browser.$(Selectors.CloseSettingsModalButton);
+  await closeButton.waitForClickable();
+  await closeButton.click();
+  await closeButton.waitForDisplayed({
+    reverse: true,
+  });
+}
+
+async function setAIOptIn(browser: CompassBrowser, enabled: boolean) {
+  // Reset the opt-in preference by using the execute command
+  await browser.setFeature('optInGenAIFeatures', enabled);
+
+  // Wait for the IPC to be processed
+  await browser.pause(500);
+}
+
+async function openAssistantDrawer(browser: CompassBrowser) {
+  const drawerButton = browser.$(Selectors.AssistantDrawerButton);
+  await drawerButton.waitForDisplayed();
+  await drawerButton.waitForClickable();
+  await drawerButton.click();
+}
+
+async function clearChat(browser: CompassBrowser) {
+  const clearChatButton = browser.$(Selectors.AssistantClearChatButton);
+  if (await clearChatButton.isDisplayed()) {
+    await clearChatButton.click();
+    const confirmButton = browser.$(
+      Selectors.AssistantConfirmClearChatModalConfirmButton
+    );
+    await confirmButton.waitForClickable();
+    await confirmButton.click();
+  }
+}
+
+async function getDisplayedMessages(browser: CompassBrowser) {
+  // Wait for the messages container to be visible
+  const chatMessages = browser.$(Selectors.AssistantChatMessages);
+  await chatMessages.waitForDisplayed();
+
+  // Get all individual message elements
+  const messageElements = await browser
+    .$$(Selectors.AssistantChatMessage)
+    .getElements();
+
+  const displayedMessages = [];
+  for (const messageElement of messageElements) {
+    const textElements = await messageElement.$$('p').getElements();
+    const isAssistantMessage =
+      textElements.length !== 1 &&
+      (await textElements[0].getText()) === 'MongoDB Assistant';
+    // Get the message text content.
+    // In case of Assistant messages, skip the MongoDB Assistant text.
+    const text = isAssistantMessage
+      ? await textElements[1].getText()
+      : await textElements[0].getText();
+
+    displayedMessages.push({
+      text: text,
+      role: isAssistantMessage ? ('assistant' as const) : ('user' as const),
+    });
+  }
+
+  return displayedMessages;
+}
+
+async function useExplainPlanEntryPoint(browser: CompassBrowser) {
+  // Open explain plan modal by clicking the explain button
+  const explainButton = browser.$(Selectors.AggregationExplainButton);
+  await explainButton.waitForDisplayed();
+  await explainButton.click();
+
+  // Wait for the explain plan modal to open and finish loading
+  const explainModal = browser.$(Selectors.AggregationExplainModal);
+  await explainModal.waitForDisplayed();
+
+  // Wait for the explain plan to be ready (loader should disappear)
```

Comment:
I would personally remove this whole block because the loader can appear/disappear instantly or it can happen within one of the intervals that any wait command polls. ie. it is going to be flaky. Plus it also suffers from the problem that the loader starts off not being there.

You can just wait for the interpret button to be there and click it - it should be the same result anyway.